temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/WorkersAutoConfiguration.java

```diff
@@ -0,0 +1,244 @@
+          Collection<Class<?>> autoDiscoveredWorkflowImplementationClasses) {
+    Set<Worker> workers =
+        properties.getWorkers().stream()
+            .map(workerProperties -> newWorker(workerFactory, workerProperties))
+            .collect(Collectors.toSet());
+
+    if (autoDiscoveredWorkflowImplementationClasses != null) {
+      for (Class<?> clazz : autoDiscoveredWorkflowImplementationClasses) {
+        WorkflowImpl annotation = clazz.getAnnotation(WorkflowImpl.class);
+        for (String taskQueue : annotation.taskQueues()) {
+          log.info(
+              "Registering auto-discovered workflow class {} on a task queue {}", clazz, taskQueue);
+          Worker worker = workerFactory.newWorker(taskQueue);
+          configureWorkflowImplementation(worker, clazz);
+          workers.add(worker);
+        }
+      }
+    }
+
+    workers.forEach(
+        worker -> beanFactory.registerSingleton("temporalWorker-" + worker.getTaskQueue(), worker));
+
+    return workers;
+  }
+
+  @ConditionalOnProperty(prefix = "spring.temporal", name = "startWorkers", matchIfMissing = true)
+  @Bean
+  public WorkerFactoryStarter workerFactoryStarter(WorkerFactory workerFactory) {
+    return new WorkerFactoryStarter(workerFactory);
+  }
+
+  Worker newWorker(WorkerFactory workerFactory, WorkerProperties workerProperties) {
+    Worker worker = workerFactory.newWorker(workerProperties.getTaskQueue());
+
+    Collection<Class<?>> workflowClasses = workerProperties.getWorkflowClasses();
+    if (workflowClasses != null) {
+      workflowClasses.forEach(
+          clazz -> {
+            log.info(
+                "Registering configured workflow class {} on a task queue {}",
+                clazz,
+                workerProperties.getTaskQueue());
+            configureWorkflowImplementation(worker, clazz);
+          });
+    }
+
+    return worker;
+  }
+
+  @SuppressWarnings("unchecked")
+  private <T> void configureWorkflowImplementation(Worker worker, Class<?> clazz) {
+    // TODO this code is a copy-past of a portion of
+    // POJOWorkflowImplementationFactory#registerWorkflowImplementationType
+    //  we should refactor it to separate the logic into reusable methods
+    boolean hasWorkflowMethod = false;
+    POJOWorkflowImplMetadata workflowMetadata = POJOWorkflowImplMetadata.newInstance(clazz);
+    for (POJOWorkflowInterfaceMetadata workflowInterface :
+        workflowMetadata.getWorkflowInterfaces()) {
+      Optional<POJOWorkflowMethodMetadata> workflowMethod = workflowInterface.getWorkflowMethod();
+      if (workflowMethod.isPresent()) {
+        // TODO this is ugly. POJOWorkflowMethodMetadata needs to be generified
+        worker.addWorkflowImplementationFactory(
+            (Class<T>) workflowInterface.getInterfaceClass(),
+            () -> (T) beanFactory.createBean(clazz));
+        hasWorkflowMethod = true;
+        break;
```

Comment:
1. This code is nasty, it will go away into the core module into the classes that do reflection lookup. It just made it work right now, this definitely will be cleaned up.